File: frontend/src/services/queueService.js
import axiosClient from "./axiosClient.js";

const queueService = {
  getQueue(filters = {}) {
    return axiosClient.get("/receptionist/queue", { params: filters });
  },

  callPatient(queueTicketId) {
    return axiosClient.put(`/receptionist/queue/${queueTicketId}/call`);
  },

  skipPatient(queueTicketId) {
    return axiosClient.put(`/receptionist/queue/${queueTicketId}/skip`);
  },

  completePatient(queueTicketId) {
    return axiosClient.put(`/receptionist/queue/${queueTicketId}/complete`);
  },

  getMyQueueStatus() {
    return axiosClient.get("/patient/queue-status");
  },
};

export default queueService;

File: backend/src/main/java/com/clinicmanagement/appointment/QueueService.java
package com.clinicmanagement.appointment;

import com.clinicmanagement.appointment.dto.PatientQueueStatusResponse;
import com.clinicmanagement.appointment.dto.QueueTicketResponse;
import java.time.LocalDate;
import java.util.List;

public interface QueueService {
    List<QueueTicketResponse> getQueue(LocalDate date, Long doctorId, String status);
    QueueTicketResponse callPatient(Long queueTicketId);
    QueueTicketResponse skipPatient(Long queueTicketId);
    QueueTicketResponse completePatient(Long queueTicketId);
    PatientQueueStatusResponse getPatientQueueStatus(Long userId);
}

